base/workerpool.jl

```diff
@@ -51,42 +67,43 @@ function remotecall_pool(rc_f, f, pool::WorkerPool, args...; kwargs...)
         if worker in procs()
             break
         else
-            pool.count = pool.count - 1
+            delete!(pool.workers, worker) # Remove invalid worker from pool
         end
     end
+    return worker
+end
 
+function remotecall_pool(rc_f, f, pool::AbstractWorkerPool, args...; kwargs...)
+    worker = take!(pool)
     try
         rc_f(f, worker, args...; kwargs...)
     finally
-        if worker != 1
-            put!(pool.channel, worker)
-        end
+        worker in pool.workers && put!(pool, worker)
```

Comment:
In case of the `default_worker_pool`, the master is considered a worker till the time new workers are added via `addprocs`. This is implicit in the sense `pool.workers` for  `default_worker_pool` will never contain pid 1.

However, it is perfectly valid for other pools to explicitly add any worker (including 1) to the pool. Hence the check.